### PR TITLE
Create user login

### PR DESCRIPTION
**Pull request by tmohaupt**

### CODE REVIEW

File: client/src/components/SignUp.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react';
+import { Button } from 'react-bootstrap';
+import { useNavigate } from 'react-router-dom';
+import { postData } from '../api';
+
+export default function Login() {
```
Comment (varangill) [inline client/src/components/SignUp.tsx]: guessing you just missed this but i'd rename this component to SignUp

File: client/src/components/NavigationBar.tsx [hunk 2]
```diff
@@ -3,17 +3,23 @@ import Nav from "react-bootstrap/Nav";
 import Navbar from "react-bootstrap/Navbar";
 import Button from "react-bootstrap/Button";
 import { Home, ClipboardCheck, ClipboardList, User2 } from 'lucide-react';
+import { useAuth } from "../AuthContext";
 
 export default function NavigationBar() {
   const navigate = useNavigate();
-  const is_loggedin = true; //TODO: login treatment
+  const { user, logout } = useAuth();
```
Comment (varangill) [inline client/src/components/NavigationBar.tsx:10]: Nitpicky but it's slightly unclear what `user` is here, could rename this to `userId` or `userObject` depending on which it is

**Review by varangill — APPROVED** [review]
LGTM, couple of minor nonblocking comments

File: client/src/components/SignUp.tsx (hunk 1 above)
Comment (tmohaupt) [reply client/src/components/SignUp.tsx]: yeah lol

File: client/src/components/NavigationBar.tsx (hunk 2 above)
Comment (tmohaupt) [reply client/src/components/NavigationBar.tsx:10]: 'user' is an object property set when the login() function is called. It should contain email:string, id:string, and name:string properties (i.e. user.name can be called in other routes this way!), but it seems that I am missing the email:string property from the interface 'User'. I'll update that in a commit before merging

**Pull request merged**